internal/apidiff/apidiff.go

```diff
@@ -0,0 +1,287 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build linux darwin
+
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"path"
+	"strings"
+)
+
+// TODO(noahdietz): remove this once the fix in golang.org/x/tools is released.
+// https://github.com/golang/go/issues/44796
+const ignored = "- MaxPublishRequestBytes: value changed from 0.000582077 to 10000000"
+const rootMod = "cloud.google.com/go"
+
+var repoMetadataPath string
+var verbose bool
+
+func init() {
+	flag.StringVar(&repoMetadataPath, "repo-metadata", "", "path to a repo-metadata-full JSON file [required]")
+	flag.BoolVar(&verbose, "verbose", false, "enable verbose command logging")
+}
+
+func main() {
+	flag.Parse()
+	if repoMetadataPath == "" {
+		log.Fatalln("Missing required flag: -repo-metadata")
+	}
+
+	run, err := shouldRun()
+	if err != nil {
+		log.Fatalln(err)
+	}
+	if !run {
+		log.Println("Not running apidiff because description contained tag BREAKING_CHANGE_ACCEPTABLE.")
+		return
+	}
+
+	d, err := newDiffer()
```

Comment:
Seems like `newDiffer` can be inlined, too.

Overall, there is a lot of indirection in `func main`, but I don't understand the names, so I don't know what is happening.

Reply:
Done. I removed most indirection.